### PR TITLE
hotfix for slow twiddles on small domain sizes

### PR DESCRIPTION
**Pull request by vhnatyk**
## Describe the changes

This PR is a hotfix for ntt domains of log2 sizes < 4

## Linked Issues

Resolves #

### CODE REVIEW

**Comment by vhnatyk** [comment]
"proper fix" - the fast twiddles must only be used with mixed radix ntt, so should be part of config and not decoupled or validated. But maybe it's late to refactor v2 for this

File: icicle/src/ntt/kernel_ntt.cu [hunk 1]
```diff
@@ -702,6 +702,14 @@ namespace mxntt {
   {
     CHK_INIT_IF_RETURN();
 
+    if (log_size < 4) {
+      // TODO: should throw exception
+      // TODO: fast twiddles should be used only with mixed radix ntt
+      // THROW_ICICLE_ERR(IcicleError_t::InvalidArgument, "log_size must be >= 4");
+      printf("warning: log_size must be at least 4, got %d\n", log_size); // TODO: logging
```
Comment (HadarIngonyama) [inline icicle/src/ntt/kernel_ntt.cu:709]: "warning: when using fast twiddles, log_size must be at least 4..

**Review by HadarIngonyama — COMMENTED** [review]
why print and not throw an exception?

**Review by yshekel — COMMENTED** [review]
I would add the following to init_domain() just before constructing the fast twiddles, to disable it from there rather than calling a function that seems to return success but doesn't do anything.

fast_twiddles_mode &= domain.max_log_size >= 4; // fast twiddles cannot be built for smaller domain

File: icicle/src/ntt/kernel_ntt.cu (hunk 1 above)
Comment (vhnatyk) [reply icicle/src/ntt/kernel_ntt.cu:709]: umm, throwing would be a breaking change

**Comment by vhnatyk** [comment]
> I would add the following to init_domain() just before constructing the fast twiddles, to disable it from there rather than calling a function that seems to return success but doesn't do anything.
> 
> fast_twiddles_mode &= domain.max_log_size >= 4; // fast twiddles cannot be built for smaller domain

so this would solve one particular call  to a `generate_external_twiddles_fast_twiddles_mode` , not fix the function itself, but since it's called only once perhaps it's indeed ok

File: icicle/src/ntt/ntt.cu [hunk 2]
```diff
@@ -469,6 +469,8 @@ namespace ntt {
       CHK_IF_RETURN(mxntt::generate_external_twiddles_generic(
         primitive_root, domain.twiddles, domain.internal_twiddles, domain.basic_twiddles, domain.max_log_size,
         ctx.stream));
+      
+      fast_twiddles_mode &= domain.max_log_size >= 4; // fast twiddles cannot be built for smaller domain
```
Comment (HadarIngonyama) [inline icicle/src/ntt/ntt.cu:473]: so the user will think he is using fast twiddles but he isn't? why not just have the exception? or at least add a print in this case

**Pull request closed without merging**